### ISSUE TITLE
textsum decode results

### ISSUE THREAD

**Issue by tumusudheer** [tensorflow/models#468]
## Interpreting textsum decode params & results

I ran textsum decode with the following parameters
`beamsize=2`
`max_decode_steps=100`
`decode_batches_per_ckpt=8`

I ran the textsum decoder with the above params for my test set that has 10 articles and summaries. When I started the decoder, I can see a couple of new files `ref<timestamp>` and `decode<timestamp>` is being generated in decode directory approximately for every 1 sec. 

Are these the files we should watch to get final output ?

Each ref<timestamp> file contains 16 sentences that are randomly selected from my test set summaries. some of these summaries are duplicated in ref<> files. A sample ref file is as follows
`
output = summary1;
output = summary2;
output = summary3;
output = summary1;
`
 A sample decoede<timestamp> file is as follows:
`
output = word1
output = 
output =
output = word2
..
`
Some of the outputs in decode files are empty. Where should I look for my final output summaries for my test articles. If I have to look in the last generated decode<> file, how should I pick 10 summaries from  16 summaries that decode<> files contains .

Am I doing anything wrong or using wrongs parameters ? Also What are `dec_timesteps` and `enc_timesteps` these parameters for ?

**Comment by tatatodd** [comment]
@panyx0718 @peterjliu might have some feedback here.  Thanks!

**Comment by goodrahstar** [comment]
Looking for the similar answers. How to interpreter the results ?

**Comment by panyx0718** [comment]
Please refer seq2seq_attention_decode.py for the details.
ref file is the original testset summaries. decode is model generated summaries.

**Comment by pranay360** [comment]
@panyx0718 Would it be possible for you to share the vocabulary and the model checkpoints trained on Gigaword dataset?

**Comment by lfzCarlosC** [comment]
have the same issue, how to check the decode result

**Comment by lfzCarlosC** [comment]
I have figured out the issue. 
If you want to check out the complete summary for all your test articles, you need to set the parameters as:
max_decode_steps=1
The reason why you get 16 summaries each file is due to parameters
beamsize=2
decode_batches_per_ckpt=8
then it output 8 batches data with batch size 2. I think this setting is for large text generation. But for you you can calculate how many number of articles your test data has and divide it by beamsize and set to  decode_batches_per_ckpt, or you can modify the code by replacing the mechanism with what you want:

for _ in xrange(FLAGS.decode_batches_per_ckpt):
      (article_batch, _, _, article_lens, _, _, origin_articles,
       origin_abstracts) = self._batch_reader.NextBatch()
      for i in xrange(self._hps.batch_size):
        ...BeamSearch...

**Comment by Zhijing1128** [comment]
Hi,

I got output=<UNK> <UNK> <UNK> <UNK> <UNK> <UNK> <UNK> <UNK> <UNK> in my decode result. How should I deal with that?

Thanks.

**Comment by karmel** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
